src/elements/dropdown/dropdown.js

```diff
@@ -0,0 +1,455 @@
+
+        if (selectedItem) {
+          selectedItem.select();
+        }
+
+        $timeout(function() {
+          $document.on('click', closeDropdown);
+        });
+      } else {
+        $document.off('click', closeDropdown);
+      }
+    });
+
+    function closeDropdown(event) {
+      self.closeDropdown(event);
+    }
+
+    $element.on('click', onClick);
+    $element.on('focus', onFocus);
+    $element.on('blur', onBlur);
+    $element.on('keydown', onKeyDown);
+
+    $element.on('mousedown', onMousedown);
+    $document.on('click', onDocumentClick);
+
+    $scope.$on('$destroy', function() {
+      $document.off('click', onDocumentClick);
+    });
+
+    function onClick(event) {
+      if (menuElm[0].contains(event.target)) {
+        return;
+      }
+      if (mouseClick) {
+        event.preventDefault();
+        self.toggle();
+      }
+
+      if (self.searchInput) {
+        self.searchInput[0].focus();
+      }
+    }
+
+    function onFocus(event) {
+      if (!mouseClick) {
+        event.preventDefault();
+        self.toggle(true);
+      }
+    }
+
+    function onBlur(event) {
+      if (!mouseClick) {
+        event.preventDefault();
+        self.toggle(false);
+      }
+    }
+
+    function onMousedown() {
+      mouseClick = true;
+    }
+
+    function onDocumentClick() {
+      mouseClick = false;
+    }
+
+    function onKeyDown(event) {
+      if (!self.getIsOpen()) {
+        switch (event.keyCode) {
+          case keys.space:
+          case keys.enter:
+          case keys.upArrow:
+          case keys.downArrow:
+            self.toggle(true);
+            break;
+        }
+      } else {
+        switch (event.keyCode) {
+          case keys.space:
+          case keys.enter:
+            var selectedItem = self.getSelectedItem();
+            if (selectedItem) {
+              selectedItem.$element.triggerHandler('click');
+            }
+            break;
+          case keys.upArrow:
+            self.selectPrev();
+            break;
+          case keys.downArrow:
+            self.selectNext();
+            break;
+          case keys.escape:
+            $element.triggerHandler('blur');
+            break;
+        }
+      }
+    }
+  }
+
+  DropdownCtrl.prototype.init = function(ngModel) {
+    var self = this;
+    self.ngModel = ngModel;
+
+    if (ngModel) {
+      ngModel.$render = function () {
+        self.items.forEach(function (item) {
+          if (item.active) {
+            self.setLabel(item.$element.html());
+            self.labelElm.removeClass('default');
+          }
+        });
+      };
+    }
+  };
+
+  DropdownCtrl.prototype.getIsOpen = function() {
+    return this.isOpen;
+  };
+
+  DropdownCtrl.prototype.setIsOpen = function(value) {
+    this.isOpen = value;
+  };
+
+  DropdownCtrl.prototype.toggle = function(open) {
+    this.setIsOpen(arguments.length ? !!open : !this.getIsOpen());
+    if (!this.$scope.$root.$$phase) {
```

Comment:
This feels like a [really bad idea](https://github.com/angular/angular.js/wiki/Anti-Patterns)